src/main/scala/com/fulcrumgenomics/umi/CodecConsensusCaller.scala

```diff
@@ -0,0 +1,196 @@
+            - longestR2Alignment.readPosAtRefPos(overlapStart, returnLastBaseIfDeleted = true)
+            !=
+            longestR1Alignment.readPosAtRefPos(overlapEnd, returnLastBaseIfDeleted = true)
+              - longestR2Alignment.readPosAtRefPos(overlapEnd, returnLastBaseIfDeleted = true)) {
+          rejectRecords((r1s.view ++ r2s.view).flatMap(_.sam), FilterIndelError)
+          Nil
+        }
+        else {
+          // Call the single-stranded consensus reads
+          val r1Consensus = this.ssCaller.consensusCall(r1s).getOrElse(unreachable("Failed to make SS consensus."))
+          val r2Consensus = this.ssCaller.consensusCall(r2s).getOrElse(unreachable("Failed to make SS consensus."))
+
+          // Re-Reverse the negative strand consensus so it's easier to line up with the other read
+          if (longestR1Alignment.negativeStrand) r1Consensus.revcomp() else r2Consensus.revcomp()
+
+          // Calculate the length of the consensus
+          computeConsensusLength(longestPosAln, longestNegAln) match {
+            case -1 =>
+              rejectRecords((r1s.view ++ r2s.view).flatMap(_.sam), FilterIndelError)
+              Nil
+            case n if n < r1Consensus.length || n < r2Consensus.length =>
+              // TODO remove this branch once SamRecordClipper has been updated
+              rejectRecords((r1s.view ++ r2s.view).flatMap(_.sam), "Temporary Issue: Fix Clipping Issue")
+              Nil
+            case consensusLength =>
+              // Pad out the vanilla consensus reads to the full length
+              val paddedR1 = r1Consensus.padded(newLength = consensusLength, left = longestR1Alignment.negativeStrand, qual = 0)
+              val paddedR2 = r2Consensus.padded(newLength = consensusLength, left = longestR2Alignment.negativeStrand, qual = 0)
+
+              val consensus = duplexConsensus(Some(paddedR1), Some(paddedR2), sourceReads = None)
+              if (longestR1Alignment.negativeStrand) consensus.foreach(_.revcomp())
+
+              val umis = recs.iterator.map(r => r[String](ConsensusTags.UmiBases)).toSeq
+              consensus
+                .iterator
+                .map(c => createSamRecord(c, ReadType.Fragment, umis))
+                .toSeq
+          }
+        }
+      }
+    }
+  }
+
+  /**
+    * Computes the length of the consensus read that should be generated. The provided positive and
+    * negative strand alignments _must_ overlap.
+    *
+    * Returns -1 if the length couldn't be computed because the overlap of the reads ends on an indel.
+    */
+  private def computeConsensusLength(pos: SamRecord, neg: SamRecord): Int = {
+    val refOverlapEnd = pos.end
+    val posReadPos = pos.readPosAtRefPos(refOverlapEnd, returnLastBaseIfDeleted=false)
+    val negReadPos = neg.readPosAtRefPos(refOverlapEnd, returnLastBaseIfDeleted=false)
+    if (posReadPos == 0 || negReadPos == 0) -1 else {
+      posReadPos + neg.length - negReadPos
+    }
```

Comment:
question: is it true that we assume that the two reads are sequenced the same # of cycles?  If so, do we have to worry about differences in lengths here and above?

Reply:
No ... there's no assumption anywhere that reads are all the same length.  The only assumption is that by the time we get here the reads have been trimmed so that they don't extend past each other - which I guess _is_ an assumption they are the same length _in the case when they are fully overlapped_.